tenant_schemas_celery/app.py

```diff
@@ -29,47 +29,61 @@ def switch_schema(task, kwargs, **kw):
     # guarantee this module was loaded when the settings were ready.
     from .compat import CompatibleConnection
 
-    compat_connection = CompatibleConnection(connection)
-    schema_name = compat_connection.get_schema().schema_name
+    default_compat_connection = CompatibleConnection(connections['default'])
+    schema_name = default_compat_connection.get_schema().schema_name
     public_schema_name = CompatibleConnection.get_public_schema_name()
 
     old_schema = schema_name
     setattr(task, "_old_schema", old_schema)
 
-    schema = get_schema_name_from_task(task, kwargs)
+    for connection in connections.all():
+        if not hasattr(connection, "_schema"):
+            continue
 
-    # If the schema has not changed, don't do anything.
-    if schema_name == schema:
-        return
+        compat_connection = CompatibleConnection(connection)
+
+        schema = get_schema_name_from_task(task, kwargs)
 
-    if schema_name != public_schema_name:
-        compat_connection.set_schema_to_public()
+        # If the schema has not changed, don't do anything.
+        if schema_name == schema:
+            continue
 
-    if schema == public_schema_name:
-        return
+        if schema_name != public_schema_name:
+            compat_connection.set_schema_to_public()
 
-    tenant = task.get_tenant_for_schema(schema_name=schema)
-    compat_connection.set_schema(tenant)
+        if schema == public_schema_name:
+            continue
+
+        tenant = task.get_tenant_for_schema(schema_name=schema)
+        compat_connection.set_schema(tenant)
 
 
 def restore_schema(task, **kwargs):
     """ Switches the schema back to the one from before running the task. """
     from .compat import CompatibleConnection
 
     public_schema_name = CompatibleConnection.get_public_schema_name()
-    compat_connection = CompatibleConnection(connection)
+
     if hasattr(task, "_old_schema"):
         schema_name = task._old_schema
```

Comment:
this is broken and makes me thing this compatibility layer is not useful (anymore?)
Maybe it's fixed by your hasattr check on connection._schema, but without, we were comparing with un-initialised schema_name